test/e2e/common/pods.go

```diff
@@ -801,4 +802,67 @@ var _ = framework.KubeDescribe("Pods", func() {
 		validatePodReadiness(false)
 
 	})
+
+	It("pod should be in running state after the graceful terimination delete call", func() {
+		logMessage := "alive still"
+		pod := getProcessedPod(podClient, f, logMessage)
+
+		By("deleting the pod")
+		err := podClient.Delete(pod.Name, metav1.NewDeleteOptions(30))
+		framework.ExpectNoError(err, "failed to delete pod")
+
+		pod, err = podClient.Get(pod.Name, metav1.GetOptions{})
+
+		By("waiting for pod running up to graceful period seconds")
+		Expect(wait.Poll(time.Second*5, 30*time.Second, func() (bool, error) {
+			pd, err := podClient.Get(pod.Name, metav1.GetOptions{})
+			if pd.Status.Phase == v1.PodRunning {
+				By("validating the pod log")
+				podLog, err := framework.GetPodLogs(f.ClientSet, f.Namespace.Name, pd.Name, pd.Spec.Containers[0].Name)
```

Comment:
Strictly speaking, the time between podClient.Get() call at line818 and framework.GetPodLogs() call at line821 can be flake if the pod is terminated between these calls.
This new test title is `"pod should be in running state after the graceful terimination delete call"`. So isn't it enough to check the pod state is running once after graceful delete call without checking the log?